modules/oe_showcase_list_pages/tests/src/ExistingSite/ListPagesTest.php

```diff
@@ -255,43 +237,194 @@ public function testCreateListPages() {
+      'oelp_oe_sc_project__end_date' => 'oelp_oe_sc_project__end_date',
+      'oelp_oe_sc_project__start_date' => 'oelp_oe_sc_project__start_date',
+      'oelp_oe_sc_project__status' => 'oelp_oe_sc_project__status',
+      'oelp_oe_sc_project__type' => 'oelp_oe_sc_project__type',
+    ]);
+
+    $this->drupalGet($list_page->toUrl());
+
+    $this->assertResultsTitle('Results', 3);
+    $this->assertResults([
+      'Project closed',
+      'Project ongoing',
+      'Project pending',
+    ]);
+
+    // Assert that the filter form for Events exists.
+    $filter_form = $assert_session->elementExists('css', '#oe-list-pages-facets-form');
+    $filter_status = $filter_form->findField('Status');
+    $filter_type = $filter_form->findField('Project type');
+    $filter_budget = $filter_form->findField('Total budget');
+    $filter_start = $filter_form->findField('Start date');
+    $filter_end = $filter_form->findField('End date');
+    $search_button = $filter_form->find('css', '#edit-submit');
+    $this->assertNotNull($filter_type);
+    $this->assertNotNull($filter_budget);
+    $this->assertNotNull($filter_start);
+    $this->assertNotNull($filter_end);
+    $this->assertNotNull($search_button);
+
+    // Filter results by type.
+    $filter_type->selectOption('financing');
+    $search_button->click();
+    $this->assertResultsTitle('Results', 1);
+    $this->assertResults([
+      'Project closed',
+    ]);
+    $filter_type->selectOption('public finance', TRUE);
+    $search_button->click();
+    $this->assertResultsTitle('Results', 2);
+    $this->assertResults([
+      'Project closed',
+      'Project pending',
+    ]);
+
+    // Filter results by budget.
+    $filter_form->findButton('Clear filters')->click();
+    $filter_budget->setValue(33);
+    $search_button->click();
+    $this->assertResultsTitle('Results', 1);
+    $this->assertResults([
+      'Project ongoing',
+    ]);
+
+    // Filter results by dates.
+    $filter_form->findButton('Clear filters')->click();
+    $filter_start->setValue('gt');
+    $date_input = $filter_form->findField('Date');
+    $date_input->setValue('2022-05-19');
+    $search_button->click();
+    $this->assertResultsTitle('Results', 2);
+    $this->assertResults([
+      'Project ongoing',
+      'Project pending',
+    ]);
+
+    $filter_start->setValue('lt');
+    $search_button->click();
+    $this->assertResultsTitle('Results', 1);
+    $this->assertResults([
+      'Project closed',
+    ]);
+
+    $dates = $filter_form->findAll('named', ['field', 'Date']);
+    $filter_start->setValue('gt');
+    $dates[0]->setValue('2022-05-19');
+    $filter_end->setValue('lt');
+    $dates[1]->setValue($date_plus_10);
+    $search_button->click();
+    $this->assertResultsTitle('Results', 1);
+    $this->assertResults([
+      'Project ongoing',
+    ]);
+
+    // Filter results by status.
+    $filter_form->findButton('Clear filters')->click();
```

Comment:
```suggestion
    $filter_form->pressButton('Clear filters');
```
This way the IDE does not need to complain about "could be NULL".

Reply:
It also throws an exception if the button is not found, which is good.